src/dagmc.cpp

```diff
@@ -37,189 +36,137 @@ const bool DAGMC_ENABLED = false;
-    if (mat_found_by_name) {
-      msg << " using material name: " << m->name_;
-    } else {
-      msg << " using material id: " << m->id_;
-    }
-    write_message(msg.str(), 10);
+  // determine the next surface id
+  int32_t next_surf_id = 0;
+  for (const auto& s : model::surfaces) {
+    if (s->id_ > next_surf_id) next_surf_id = s->id_;
   }
-}
+  surf_idx_offset_ = model::surfaces.size();
+  next_surf_id++;
 
-void load_dagmc_geometry()
-{
-  if (!model::DAG) {
-    model::DAG = new moab::DagMC();
-  }
+  // create a new DAGMC instance
+  dagmc_instance_ = std::make_shared<moab::DagMC>();
 
   // --- Materials ---
 
-  // create uwuw instance
-  auto filename = dagmc_file();
-  UWUW uwuw(filename.c_str());
+  // read any UWUW materials from the file
+  read_uwuw_materials();
 
   // check for uwuw material definitions
-  bool using_uwuw = !uwuw.material_library.empty();
+  bool using_uwuw = uses_uwuw();
 
   // notify user if UWUW materials are going to be used
   if (using_uwuw) {
     write_message("Found UWUW Materials in the DAGMC geometry file.", 6);
   }
 
-  int32_t dagmc_univ_id = 0; // universe is always 0 for DAGMC runs
-
   // load the DAGMC geometry
-  moab::ErrorCode rval = model::DAG->load_file(filename.c_str());
+  filename_ = settings::path_input + filename_;
+  if (!file_exists(filename_)) {
+    fatal_error("Geometry DAGMC file '" + filename_ + "' does not exist!");
+  }
+  moab::ErrorCode rval = dagmc_instance_->load_file(filename_.c_str());
   MB_CHK_ERR_CONT(rval);
 
   // initialize acceleration data structures
-  rval = model::DAG->init_OBBTree();
+  rval = dagmc_instance_->init_OBBTree();
   MB_CHK_ERR_CONT(rval);
 
   // parse model metadata
-  dagmcMetaData DMD(model::DAG, false, false);
+  dagmcMetaData DMD(dagmc_instance_.get(), false, false);
   DMD.load_property_data();
 
   std::vector<std::string> keywords {"temp"};
   std::map<std::string, std::string> dum;
   std::string delimiters = ":/";
-  rval = model::DAG->parse_properties(keywords, dum, delimiters.c_str());
+  rval = dagmc_instance_->parse_properties(keywords, dum, delimiters.c_str());
   MB_CHK_ERR_CONT(rval);
 
   // --- Cells (Volumes) ---
 
   // initialize cell objects
-  int n_cells = model::DAG->num_entities(3);
+  int n_cells = dagmc_instance_->num_entities(3);
   moab::EntityHandle graveyard = 0;
   for (int i = 0; i < n_cells; i++) {
-    moab::EntityHandle vol_handle = model::DAG->entity_by_index(3, i+1);
+    moab::EntityHandle vol_handle = dagmc_instance_->entity_by_index(3, i + 1);
 
     // set cell ids using global IDs
-    DAGCell* c = new DAGCell();
-    c->dag_index_ = i+1;
-    c->id_ = model::DAG->id_by_index(3, c->dag_index_);
-    c->dagmc_ptr_ = model::DAG;
-    c->universe_ = dagmc_univ_id; // set to zero for now
+    auto c = std::make_unique<DAGCell>();
+    c->dag_index_ = i + 1;
+    c->id_ = adjust_geometry_ids_ ? next_cell_id++ : dagmc_instance_->id_by_index(3, c->dag_index_);
```

Comment:
say i tally in cell 3, which is a dagmc cell, do we accumulate tally scores across all the instances of that cell individually, or would I get a score for each 'copy'?

Reply:
When using a `CellFilter` you'll get the accumulated score over all the instances of that cell in the geometry. If you use a `CellInstanceFilter` filter or a `DistribCellFilter` you can get for each instance of the cell in the geometry.